engine/runtime-compiler/src/main/scala/org/enso/compiler/Compiler.scala

```diff
@@ -816,8 +816,8 @@ class Compiler(
       .diagnostics
     val module    = inlineContext.getModule()
     val hasErrors = reportDiagnostics(errors, module)
-    if (hasErrors && inlineContext.compilerConfig.isStrictErrors) {
-      throw context.throwAbortedException()
+    if (hasErrors.nonEmpty && inlineContext.compilerConfig.isStrictErrors) {
+      throw hasErrors.head
     }
```

Comment:
```suggestion
    hasErrors match {
        case error :: _ if inlineContext.compilerConfig.isStrictErrors =>
            throw error
        case _ =>
    }
```
would be more Scala'ish to write it